### DISCUSSION TITLE
Why javascript function doesn't work in leetcode?

### DISCUSSION THREAD

**Discussion by KhambhanSingh** [community/community#61139] [Programming Help]
### Hi everyone,

var twoSum = function(l1,l2) {
        var num1 = l1.reverse().join('');
        var num2 = l2.reverse().join('');
        sum = parseInt(num1)+parseInt(num2);
        var myArr = String(sum).split("").map((sum)=>{
            return Number(sum)
            });
    return myArr.reverse();
    };
    function valued(){
        console.log(twoSum([9,9,9,9,9,9,9],[9,9,9,9]));
    }

Why reverse function is not working in leetcode but it works on default javascript.
[enter image description here](https://i.stack.imgur.com/4GuR0.png)

**Comment by alirezarastineh** [comment]
The JavaScript function you provided appears to be correct syntactically. However, there could be several reasons why it may not work as expected in LeetCode:

1. Function Signature: LeetCode may require a specific function signature for the problem you are attempting to solve. Make sure the function name, parameter names, and the return statement match the problem's requirements.

2. Input Format: LeetCode often provides input in a different format than the one you have assumed. Ensure that you are handling the input correctly based on the problem's specifications.

3. Library Restrictions: LeetCode may have restrictions on using certain JavaScript libraries or features. Check the problem's guidelines to see if there are any limitations on the JavaScript language features you can use.

4. Execution Environment: LeetCode may have its own execution environment with specific configurations. Ensure that your code does not rely on any external dependencies that might not be available in the LeetCode environment.

To better understand the issue, it would be helpful if you could provide more information, such as the specific problem you are trying to solve on LeetCode and any error messages or unexpected behavior you encounter.

**Comment by rumon5h** [comment]
var twoSum = function(l1,l2) {
var num1 = l1.reverse().join('');
var num2 = l2.reverse().join('');
sum = parseInt(num1)+parseInt(num2);
var myArr = String(sum).split("").map((sum)=>{
return Number(sum)
});
return myArr.reverse();
};
function valued(){
console.log(twoSum([9,9,9,9,9,9,9],[9,9,9,9]));
}


You have to call the function from below the function

valued();

**Comment by samus-aran** [comment]
Welcome to the GitHub Community, @KhambhanSingh , we're happy you're here! You are more likely to get a useful response if you are posting your question(s) in the applicable category and are explicit about what your project entails--giving a few more details might help someone give you a nudge in the right direction. I've gone ahead and moved it for you. Good luck!
